cppapi/server/multiattribute.cpp

```diff
@@ -786,26 +802,38 @@ void MultiAttribute::add_attribute(string &dev_name,DeviceClass *dev_class_ptr,l
 	{
 		if (idl_3 == false)
 		{
-			attr_list.push_back(new WAttribute(prop_list,attr,dev_name,index));
+			Attribute * new_attr = new WAttribute(prop_list,attr,dev_name,index);
+			add_attr(new_attr);
 			index = attr_list.size() - 1;
 		}
 		else
 		{
-			attr_list.insert(ite,new WAttribute(prop_list,attr,dev_name,index));
+			Attribute * new_attr = new WAttribute(prop_list,attr,dev_name,index);
```

Comment:
Seems to me as a code duplication. Must be extracted into a dedicated method overloaded from add_attr, see below